flocker/node/agents/test/test_blockdevice.py

```diff
@@ -1164,6 +1196,51 @@ def setUp(self):
     return Tests
 
 
+class IBlockDeviceAsyncAPITestsMixin(object):
+    """
+    Tests to perform on ``IBlockDeviceAsyncAPI`` providers.
+    """
+    def test_interface(self):
+        """
+        The API object provides ``IBlockDeviceAsyncAPI``.
+        """
+        self.assertTrue(
+            verifyObject(IBlockDeviceAsyncAPI, self.api)
+        )
+
+
+def make_iblockdeviceasyncapi_tests(blockdeviceasync_api_factory):
+    """
+    :return: A ``TestCase`` with tests that will be performed on the supplied
+        ``IBlockDeviceAsyncAPI`` provider.  These tests are not exhaustive
+        because we currently assume ``make_iblockdeviceapi_tests`` will be used
+        on the wrapped object.
+    """
```

Comment:
Probably not.  Since `auto_threaded` changes the interface of the class it decorates (makes it return `Deferred`s instead of synchronous results) any application code that wants to use the decorated class needs to be aware of this.  The tests are all written against a synchronous interface now.  They'll need to be rewritten to account for the new asynchronous interface.

It's possibly the rewritten result will be usable against both `IBlockDeviceAsyncAPI` and `IBlockDeviceAPI` (if we're a little careful in how we rewrite them - but it's kind of easy to support sync APIs in code written for async APIs - at least, easier than supporting async APIs in code written for sync APIs).  I'm not sure yet if that's the way to go.  It's appealing because having a single set of tests to run against all the implementations is a nice way to prove they all have similar behavior and it's probably less code to maintain.  On the other hand, the naive, synchronous assumptions made in the current tests make the failures a little bit nicer to read and debugging a little simpler.

A bridge to cross when we reach it, I think.